File: app/src/main/java/com/riders/thelab/ui/mainactivity/ViewPager2Adapter.java
package com.riders.thelab.ui.mainactivity;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import java.util.List;

public class ViewPager2Adapter extends FragmentStateAdapter {

    private List<Fragment> fragmentList;

    public ViewPager2Adapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle, List<Fragment> fragmentList) {
        super(fragmentManager, lifecycle);

        this.fragmentList = fragmentList;
    }


    public void addFragment(Fragment fragment) {
        fragmentList.add(fragment);
    }

    @Override
    public int getItemCount() {
        if (null != fragmentList && !fragmentList.isEmpty())
            return fragmentList.size();

        return 0;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return fragmentList.get(position);
    }
}

File: app/src/main/java/com/riders/thelab/ui/mainactivity/ViewPager2Adapter.kt
package com.riders.thelab.ui.mainactivity

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import kotlinx.coroutines.DelicateCoroutinesApi

class ViewPager2Adapter @OptIn(DelicateCoroutinesApi::class) constructor(
    val activity: MainActivity,
    private val fragmentList: List<Fragment>
) : FragmentStateAdapter(activity) {

    private val mFragmentList: MutableList<Fragment> = fragmentList as MutableList<Fragment>

    fun addFragment(fragment: Fragment) {
        mFragmentList.add(fragment)
    }

    fun getFragment(position: Int): Fragment {
        return mFragmentList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemCount(): Int {
        return if (mFragmentList.isNotEmpty()) mFragmentList.size else 0
    }

    override fun createFragment(position: Int): Fragment {
        return mFragmentList[position]
    }
}